### PR TITLE
Vim-script to lua

### PR DESCRIPTION
**Pull request by LalleSX**
Converted vim script to lua script. Its a draft due to i think it needs to be approved that everything that is written is working and most optimal.

## Why?

Well lua just works. 

It's syntax is easy and that makes it easily scrip table.

I left the original vim script file for now to use as a checklist to see if everything is set correctly.

### CODE REVIEW

**Comment by LalleSX** [comment]
>Its a draft ...

I pressed the wrong button...

File: .config/nvim/init.lua [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+o.filetype = "plugin"
+o.syntax = "on"
+o.encoding = "utf-8"
+o.number = true
+o.relativenumber = true
+
+--Enable autocompletion:
+o.wildmode = "longest,list,full"
+--Disables automatic commenting on newline:
+cmd("autocmd FileType * setlocal formatoptions-=c formatoptions-=r formatoptions-=o")
```
[the remark this answers is not in the document]
Comment (LalleSX) [reply .config/nvim/init.lua]: I have now made all autocmd's use the vim.api function that you showed here.
I made it in a local function so its easier to type


```lua
local function autocmd(event, pattern, cmd)
    vim.api.nvim_create_autocmd(event, {
        pattern = pattern,
        command = cmd,
    })
end
```

Use example

```lua
autocmd("BufWritePost", "config.h", "!cd ~/.local/src/dwmblocks/; sudo make install && { killall -q dwmblocks;setsid -f dwmblocks }")
```
>Overall I'm seeing a lot of unnecessary Vim script.

Can you clarify what you mean by this? So i can make it better

File: .config/nvim/init.lua [hunk 2]
```diff
@@ -0,0 +1,176 @@
+-- Make local variables to shorten the code
+local g = vim.g
+local o = vim.o
+local A = vim.api
+local fn = vim.fn
+local cmd = vim.cmd
+local b = vim.b
+local opt = vim.opt
```
[the remark this answers is not in the document]
Comment (LalleSX) [reply .config/nvim/init.lua:8]: Typing vim.x again and again, is annoying

I guess either the constants can be renamed or replaced when its fully ready to merge

File: .config/nvim/init.lua (hunk 1 above)
[the remark this answers is not in the document]
Comment (dawidpotocki) [reply .config/nvim/init.lua]: Usage of `cmd`.

**Comment by LalleSX** [comment]
Update:
- I removed all the local constants/variables and made a function that loads the settings instead.

- Usage of `vim.cmd` is now only when there wasn't any alternative

- Map is now a constant instead of a function. This uses a constant `s` to set the map to `silent`

The top of the file is a bit ugly with the functions, but it has to be there as lua requires constants & functions to be declared first before being used.

An alternative to that is something like creating two folders and file `lua/utils/utils.lua` that contains the functions and then have 
```lua 
require("utils.utils)
```  
in the `init.lua` file.

In my opinion its ready to be merged.

**Pull request closed without merging**